src/Tvheadend.cpp

```diff
@@ -1385,13 +1385,9 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   {
     UPDATE(rec.state, PVR_TIMER_STATE_COMPLETED);
   }
-  else if (strstr(state, "missed") != NULL)
-  {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
-  }
-  else if (strstr(state, "invalid")   != NULL)
+  else
   {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
+    UPDATE(rec.error, state);
```

Comment:
Assigning the state provided by tvh ("completed", "missed", ...) to the string error makes no sense for me.

Please keep this lines as they are. NO change here, please.